internal/controller/template_controller_test.go

```diff
@@ -60,87 +60,144 @@ var _ = Describe("Template Controller Integration Test", func() {
 			return fmt.Errorf("namespace not active yet")
 		}, 10*time.Second, 2*time.Second).Should(Succeed())
 
-	})
-
-	AfterEach(func() {
-		By("deleting all applied resources")
+		By("creating input resources")
 		exampleFolderPath := path.Join("../", "../", "examples")
 		exampleFolders, err := utils.GetSubDirs(exampleFolderPath)
 		Expect(err).NotTo(HaveOccurred())
 		for _, exampleFolder := range exampleFolders {
-			deleteExampleResources(ctx, path.Join(exampleFolderPath, exampleFolder, "input"))
+			inputFolder := path.Join(exampleFolderPath, exampleFolder, "input")
+			templateFolder := path.Join(inputFolder, "templates")
+			applyYAMLFilesFromDirectory(ctx, inputFolder, []string{templateFolder})
 		}
-
 	})
 
-	It("should apply and verify examples", func() {
-
+	AfterEach(func() {
+		By("deleting input resources and templates")
 		exampleFolderPath := path.Join("../", "../", "examples")
 		exampleFolders, err := utils.GetSubDirs(exampleFolderPath)
 		Expect(err).NotTo(HaveOccurred())
 		for _, exampleFolder := range exampleFolders {
-			By(fmt.Sprintf("Running example %s", exampleFolder))
-			applyExampleResources(ctx, path.Join(exampleFolderPath, exampleFolder))
+			inputFolder := path.Join(exampleFolderPath, exampleFolder, "input")
+			deleteYAMLFilesFromDirectory(ctx, inputFolder)
 		}
+	})
 
-		// another eventuallly block to check if the output.yaml file has been created
-		By("verifying that the output.yaml file has been created")
-		for _, exampleFolder := range exampleFolders {
-			exampleFolderPath := path.Join("..", "..", "examples", exampleFolder)
-			verifyExampleOutput(exampleFolderPath, "out.yaml")
-		}
+	Context("When Template is applied", func() {
+		It("creates expected managed resources", func() {
+			exampleFolderPath := path.Join("../", "../", "examples")
+			exampleFolders, err := utils.GetSubDirs(exampleFolderPath)
+			Expect(err).NotTo(HaveOccurred())
 
+			for _, exampleFolder := range exampleFolders {
+				By(fmt.Sprintf("Running example %s", exampleFolder))
+
+				By("applying example templates")
```

Comment:
Hmm, I'm not sure about this one. Here's what I'm thinking: 

In the `BeforeEach` of "When Template is deleted", we're looping to apply the templates from all of the examples and not doing anything else in the loop. So the `By` can go outside of the loop since it doesn't matter which iteration we're on. 

However, for the tests in "When Template is applied", we're looping to apply each example's template(s) and then  further verifying that the Template has the expected behaviour for that example. So the `By` being in the loop, gives us more context on the example/iteration we're on. 

So I think these make sense being in the loop, but please let me know if I'm missing something/misunderstanding the review!

Reply:
Got it; makes sense.